hipcheck-sdk-macros/src/lib.rs

```diff
@@ -0,0 +1,257 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use convert_case::Casing;
+use proc_macro::TokenStream;
+use proc_macro2::Span;
+use std::ops::Not;
+use std::sync::{LazyLock, Mutex};
+use syn::spanned::Spanned;
+use syn::{parse_macro_input, Error, Ident, ItemFn, Meta, PatType};
+
+static QUERIES: LazyLock<Mutex<Vec<NamedQuerySpec>>> = LazyLock::new(|| Mutex::new(vec![]));
+
+#[allow(unused)]
+#[derive(Debug, Clone)]
+struct NamedQuerySpec {
+	pub struct_name: String,
+	pub function: String,
+	pub default: bool,
+}
+
+struct QuerySpec {
+	pub function: Ident,
+	pub input_type: syn::Type,
+	pub output_type: syn::Type,
+	pub default: bool,
+}
+
+/// Parse Path to confirm that it represents a Result<T: Serialize> and return the type T
+fn parse_result_generic(p: &syn::Path) -> Result<syn::Type, Error> {
+	use syn::GenericArgument;
+	use syn::PathArguments;
+	// Assert it is a Result
+	let first = p.segments.first().unwrap();
```

Comment:
This will fail if the return type Path is `hipcheck_sdk::prelude::Result` instead of just `Result`.
```
error: Expected return type to be a Result<T: Serialize>
  --> plugins/dummy_sha256_sdk/src/main.rs:22:6
   |
22 | ) -> hipcheck_sdk::prelude::Result<Vec<u8>> {
```

I believe you want use `segments.last()` to get the type-name part of the path